oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java

```diff
@@ -1644,11 +1647,39 @@ public List<Facet> getFacets(int numberOfFacets, String columnName) throws IOExc
                 return cachedResults.get(cacheKey);
             }
             LOG.trace("columnName = {} facet Data not present in cache...", columnName);
+            if (EAGER_FACET_CACHE_FILL) {
```

Comment:
As we are filling cache with all facets. Would it make sense to initialise it in constructor itself using future. Like @fabriziofortino did in https://github.com/apache/jackrabbit-oak/pull/1217

Reply:
I prefer to avoid concurrency. it is filled when the first one is accessed, so I think it "should be fine". (I know, it sounds like "famous last words")